### DISCUSSION TITLE
Cpptools defaults are not in effect while working over Remote SSH

### DISCUSSION THREAD

**Discussion by aol-nnov** [microsoft/vscode-cpptools#11923] [General]
Hi!

I've started with idea to create vscode C++ profile with sensible defaults for most of my projects, so, I stuffed my `settings.json` with `C_Cpp.default.*` lines.

Imagine my surprise after connecting to remote when I saw greyed-out params in my `settings.json` and a popup `This setting cannot be applied in this window. It will be applied when you open a local window.`

Any chances to make in work with remote windows?

Thanks in advance,
Andrey

**Comment by Colengms** [comment]
Hi @aol-nnov .  Some settings, such as a compiler path, will likely differ between machines (and platforms), so are set internally as `machine-overridable`. This distinction ensure potentially invalid platform-specific settings are not automatically shared to a machine they won't apply to. https://code.visualstudio.com/api/references/contribution-points#:~:text=machine%2Doverridable%20%2D%20Machine%20specific%20settings,%2C%20workspace%2C%20or%20remote%20settings.

When using a WSL instance, there is a set of setting global for that instance.

![image](https://github.com/microsoft/vscode-cpptools/assets/49173979/7255230b-fc78-4e04-8c5c-7e3b074db31d)

From the Config UI, you can also go to the JSON file using the button here:

![image](https://github.com/microsoft/vscode-cpptools/assets/49173979/09b271df-7d9d-4671-9647-2e8786451d6d)

That should open, i.e. : 

![image](https://github.com/microsoft/vscode-cpptools/assets/49173979/6ef010e2-dbb6-468a-9a77-9c92f0ef5a34)
